teos/src/bitcoin_cli.rs

```diff
@@ -129,8 +132,10 @@ impl<'a> BitcoindClient<'a> {
     /// Gets a fresh RPC client.
     pub fn get_new_rpc_client(&self) -> std::io::Result<RpcClient> {
         let http_endpoint = HttpEndpoint::for_host(self.host.to_owned()).with_port(self.port);
-        let rpc_credentials = base64::encode(&format!("{}:{}", self.rpc_user, self.rpc_password));
-        RpcClient::new(&rpc_credentials, http_endpoint)
+        let binding =
+            general_purpose::URL_SAFE.encode(format!("{}:{}", self.rpc_user, self.rpc_password));
+        let rpc_credentials = binding.as_str();
+        Ok(RpcClient::new(&rpc_credentials, http_endpoint))
```

Comment:
no need for the `.as_str()` line.
`&String` will be tread as `&str` by the compiler.